include/aws/mqtt/client.h

```diff
@@ -161,6 +161,8 @@ typedef void(aws_mqtt_client_publish_received_fn)(
 /** Called when a connection is closed, right before any resources are deleted */
 typedef void(aws_mqtt_client_on_disconnect_fn)(struct aws_mqtt_client_connection *connection, void *userdata);
 
+typedef void(aws_mqtt_client_on_termination_fn)(struct aws_mqtt_client_connection *connection, void *userdata);
```

Comment:
The termination callback probably should not take the `connection` as a parameter, as the connection might be destroyed or at least would be unaccessible soon.